File: src/Plotly.NET/Traces/ObjectAbstractions/Icicle.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

type IcicleRoot() =
    inherit DynamicObj()

    static member init(?Color: Color) =
        IcicleRoot() |> IcicleRoot.style (?Color = Color)

    static member style(?Color: Color) =
        fun (icicleRoot: IcicleRoot) ->
            icicleRoot
            |> DynObj.withOptionalProperty "color" Color

type IcicleLeaf() =
    inherit DynamicObj()

    static member init(?Opacity: float) =
        IcicleLeaf() |> IcicleLeaf.style (?Opacity = Opacity)

    static member style(?Opacity: float) =
        fun (icicleLeaf: IcicleLeaf) ->
            icicleLeaf
            |> DynObj.withOptionalProperty "opacity" Opacity

type IcicleTiling() =
    inherit DynamicObj()

    static member init
        (
            ?Flip: StyleParam.TilingFlip,
            ?Orientation: StyleParam.Orientation,
            ?Pad: int
        ) =
        IcicleTiling() |> IcicleTiling.style (?Flip = Flip, ?Orientation = Orientation, ?Pad = Pad)

    static member style
        (
            ?Flip: StyleParam.TilingFlip,
            ?Orientation: StyleParam.Orientation,
            ?Pad: int
        ) =
        fun (icicleTiling: IcicleTiling) ->

            icicleTiling
            |> DynObj.withOptionalPropertyBy "flip" Flip StyleParam.TilingFlip.convert
            |> DynObj.withOptionalPropertyBy "orientation" Orientation StyleParam.Orientation.convert
            |> DynObj.withOptionalProperty "pad" Pad

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Icicle.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.IcicleRoot InitIcicleRoot
        (
            Optional<Color> Color = default
        ) 
            =>
                Plotly.NET.TraceObjects.IcicleRoot.init(
                    Color: Color.ToOption()
                );

    public static Plotly.NET.TraceObjects.IcicleLeaf InitIcicleLeaf
        (
            Optional<double> Opacity = default
        )
            =>
                Plotly.NET.TraceObjects.IcicleLeaf.init(
                    Opacity: Opacity.ToOption()
                );

    public static Plotly.NET.TraceObjects.IcicleTiling InitIcicleTiling
        (

            Optional<StyleParam.TilingFlip> Flip = default,
            Optional<StyleParam.Orientation> Orientation = default,
            Optional<int> Pad = default
        ) 

            =>
            Plotly.NET.TraceObjects.IcicleTiling.init(

                Flip: Flip.ToOption(),
                Orientation: Orientation.ToOption(),
                Pad: Pad.ToOption()
            );
}

